src/completion.ts

```diff
@@ -0,0 +1,118 @@
+import { Arg, Generator, Option, Spec, Subcommand } from "./types.ts";
+import { makeArray } from "./util.ts";
+
+export const JSON_START = "----- Begin JSON -----\n";
+export const JSON_END = "\n----- End JSON -----";
+
+export const completion: Subcommand = {
+  name: "completion",
+  description: "Generate Fig completions",
+  subcommands: [
+    {
+      name: "fig",
+      description: "Generates Fig completions",
+      async action() {
+      },
+    },
+    {
+      name: "run-generator",
+      description: "Runs a generator",
+      args: [
+        {
+          name: "lookup-path",
+          description: `The lookup path of the generator`,
+        },
```

Comment:
Currently expects a string like `s:0,s:1,o:3,a:0,g:0`, this needs to be documented.

Would also be fun to make a generator for this string 😄